### ISSUE TITLE
Include 'radiogroup' role to the list of selectors/scope in the aria-toggle-field-name rule

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#3256]
Add 'radiogroup' role to the list of selectors in the aria-toggle-field-name rule

Product: axe-core

Expectation: Fail elements with role ```radiogroup``` and without an accessible label

Actual: ```radiogroup``` elements without an accessible label are not failed

Motivation: Ensure the purpose of toggle field is accessible to all users

<pre><code> axe-core version: 4.3.5 </code></pre>

**Comment by WilcoFiers** [comment]
@iamrafan While ARIA does say radiogroups must have an accessible name, I don't think WCAG does. There are a good number of situations where context and good names for individual radiobuttons is sufficient to make the purpose clear. There are no other groups in ARIA or WCAG that require an accessible name. This one doesn't seem correct to us. I'm inclined to leave it as is.

**Comment by scottaohara** [comment]
@WilcoFiers I agree with your thinking here, as there are definitely situations where a radio group (using a fieldset or role=radiogroup) do not need an explicit name if 
1. there is no visible label that would be expected to be the group's name. (if there was a visible label though, then we're looking at a 1.3.1 issue, but that might be hard for automation to determine?)
2. the radio labels themselves provide enough context where a group name would be redundant.

Might a rule like the [ARIA dialog name rule](https://github.com/dequelabs/axe-core/blob/0677565941486cf339e7267760d4e533d4a29a05/lib/rules/aria-dialog-name.json) be a suitable solution here?  A best practice to indicate that [ARIA requires the `radiogroup` has a name](https://www.w3.org/TR/wai-aria-1.1/#radiogroup), but yes, not necessarily a WCAG failure per the nuance of the above two points?

**Comment by WilcoFiers** [comment]
Since I received no objections, I'm going to close the issue. If anyone cares to revisit they're free to open a new issue.
